subway_structure.py

```diff
@@ -1058,6 +1062,16 @@ def process_tracks(self, stop_position_elements):
             self.is_circular = (
                 self.stops[0].stoparea == self.stops[-1].stoparea
             )
+            if (
+                self.is_circular
+                and len(self.tracks) > 0
```

Comment:
This will be enough to check, whether the list is not empty:
```suggestion
                and self.tracks
```

Reply:
\+